recipes/llvm/all/conanfile.py

```diff
@@ -0,0 +1,128 @@
+    license = 'Apache-2.0'
+    topics = 'cpp', 'compiler', 'tooling', 'clang'
+
+    settings = 'os', 'arch', 'compiler', 'build_type'
+
+    no_copy_source = True
+    _source_subfolder = 'source_subfolder'
+
+    options = {**{ 'with_' + project : [True, False] for project in projects }, **{
+        'fPIC': [True, False]
+    }}
+    default_options = {**{ 'with_' + project : project in default_projects for project in projects }, **{
+        'fPIC': True
+    }}
+    generators = 'cmake_find_package'
+
+    @property
+    def repo_folder(self):
+        return os.path.join(self.source_folder, self._source_subfolder)
+
+    def project_folder(self, project):
+        return os.path.join(self.repo_folder, project)
+
+    def source(self):
+        tools.get(**self.conan_data["sources"][self.version])
+        extracted_dir = 'llvm-project-llvmorg-' + self.version
+        os.rename(extracted_dir, self._source_subfolder)
+
+    def build_requirements(self):
+        self.build_requires("cmake/3.21.3")
+
+    def configure(self):
+        if self.settings.compiler.get_safe("cppstd"):
+            tools.check_min_cppstd(self, '14')
+
+    def _cmake_configure(self):
+        enabled_projects = [project for project in projects if getattr(self.options, 'with_' + project)]
+        self.output.info('Enabled LLVM subprojects: {}'.format(', '.join(enabled_projects)))
+
+        cmake = CMake(self, parallel=False);
+        cmake.configure(
+            defs = {
+                'LLVM_ENABLE_PROJECTS': ';'.join(enabled_projects),
+                'LLVM_ENABLE_BINDINGS': False,
+            },
+            source_folder = os.path.join(self._source_subfolder, 'llvm')
+        )
+        return cmake
+
+    def build(self):
+        cmake = self._cmake_configure()
+        cmake.build()
+
+    def package(self):
+        cmake = self._cmake_configure()
+        cmake.install()
+
+        self.copy(
+            'LICENSE.TXT',
+            src=self.project_folder('clang'),
+            dst='licenses',
+            keep_path=False)
+
+        ignore = [
+            'share',
+            'libexec',
+            '**/Find*.cmake',
+            '**/*Config.cmake'
+        ]
+
+        for ignore_entry in ignore:
+            ignore_glob = os.path.join(self.package_folder, ignore_entry)
+
+            for ignore_path in glob.glob(ignore_glob, recursive=True):
+                self.output.info('Remove ignored file/directory "{}" from package'.format(ignore_path))
+
+                if os.path.isfile(ignore_path):
+                    os.remove(ignore_path)
+                else:
+                    shutil.rmtree(ignore_path)
+
+    def validate(self):
+        if self.settings.compiler == "gcc" and tools.Version(self.settings.compiler.version) < "10":
+            raise ConanInvalidConfiguration("Compiler version too low for this package.")
+
+        if self.settings.compiler == "Visual Studio" and Version(self.settings.compiler.version) < "16.4":
```

Comment:
Now, to get ready Conan 2.0, keep in mind that "msvc" is the "new" name for VS compilers. 
Use `(self.settings.compiler == "Visual Studio" or self.settings.compiler == "msvc")`

Reply:
Thanks, I changed this